src/FileType_XDMF.f90

```diff
@@ -0,0 +1,1236 @@
+    ENDDO
+  ENDIF
+  IF( ALLOCATED(thatmesh%material_ids) )THEN
+    IF(ALLOCATED(thismesh%material_ids)) DEALLOCATE(thismesh%material_ids)
+    ALLOCATE(thismesh%material_ids(SIZE(thatmesh%material_ids)))
+    thismesh%material_ids = thatmesh%material_ids
+  ENDIF
+  IF( ALLOCATED(thatmesh%cell_sets) ) THEN
+    IF(ALLOCATED(thismesh%cell_sets))THEN
+      DO i=1, SIZE(thismesh%cell_sets)
+        DEALLOCATE(thismesh%cell_sets(i)%cell_list)
+      ENDDO
+      DEALLOCATE(thismesh%cell_sets)
+    ENDIF
+    ALLOCATE(thismesh%cell_sets(SIZE(thatmesh%cell_sets)))
+    DO i = 1, SIZE(thatmesh%cell_sets)
+      ALLOCATE(thismesh%cell_sets(i)%cell_list(SIZE(thatmesh%cell_sets(i)%cell_list)))
+      thismesh%cell_sets(i)%cell_list = thatmesh%cell_sets(i)%cell_list
+      thismesh%cell_sets(i)%name = thatmesh%cell_sets(i)%name
+    ENDDO
+  ENDIF
+ENDSUBROUTINE assign_XDMFMeshType
+!
+!-------------------------------------------------------------------------------
+!> @brief Export the leaf nodes of the mesh hierarchy
+!> @param mesh the mesh
+!> @param xmle the XML element
+!> @param strpath the string holding the path to the XDMF file
+!> @param h5 the HDF5 file
+!>
+RECURSIVE SUBROUTINE export_leaf_XDMFFileType(mesh, xmle, strpath, h5)
+  CHARACTER(LEN=*),PARAMETER :: myName='export_leaf_XDMFFileType'
+  TYPE(XDMFMeshType),INTENT(IN)  :: mesh
+  TYPE(XMLElementType),TARGET,INTENT(INOUT) :: xmle
+  TYPE(StringType),INTENT(INOUT) :: strpath
+  TYPE(HDF5FileType), INTENT(INOUT) :: h5
+  TYPE(XMLElementType),POINTER :: current_xml, child_xml
+  TYPE(StringType) :: str_name, str_value, str1, str2, toponame, xdmf_id_str 
+  INTEGER(SNK) :: nchildren, child_ctr
+  INTEGER(SLK) :: xdmf_id, nverts, ncells, i, j, vctr
+  CHARACTER(LEN=200) :: charpath
+  INTEGER(SLK), ALLOCATABLE :: vertex_list_2d(:, :), vertex_list_1d(:), cell_list_1d(:)
+
+  ! Create HDF5 group
+  CALL h5%mkdir(CHAR(mesh%name))
+
+  ! Determine number of xmle children
+  ! Geometry, Topology, Material, Cell sets
+  ! Assumes mesh has geometry and topology
+  nchildren = 2
+  IF(ALLOCATED(mesh%material_ids)) nchildren = nchildren + 1
+  IF(ALLOCATED(mesh%cell_sets)) nchildren = nchildren + SIZE(mesh%cell_sets)
+  ALLOCATE(xmle%children(nchildren))
+
+  ! GEOMETRY
+  current_xml => xmle%children(1)
+  str_name="Geometry"
+  CALL current_xml%setName(str_name) 
+  current_xml%nAttr=0
+  current_xml%parent => xmle
+
+  str_name= "GeometryType"
+  str_value = "XYZ"
+  CALL current_xml%setAttribute(str_name, str_value)
+
+  ALLOCATE(current_xml%children(1))
+  child_xml => current_xml%children(1)
+  str_name="DataItem"
+  CALL child_xml%setName(str_name) 
+  child_xml%nAttr=0
+  child_xml%parent => current_xml
+
+  str_name= "DataType"
+  str_value = "Float"
+  CALL child_xml%setAttribute(str_name, str_value)
+
+  nverts=SIZE(mesh%vertices, DIM=2)
+  str_name="Dimensions"
+  str1 = nverts
+  str2 = "3"
+  str_value = str1//" "//str2
```

Comment:
Can this be consolidated to one line without str1 and str2?

Reply:
Not without changing to a formatted write statement into a character array. The str1 = nverts is an assignment from integer to string. Strings have the // operator defined, so this was easiest for me.